bibliopixel/return_codes.py

```diff
@@ -29,5 +29,5 @@ class BiblioSerialError(Exception):
 
 def print_error(error, exception_class=BiblioSerialError):
     msg = RETURN_CODE_ERRORS.get(error, 'Unknown error occured.')
-    log.error('%s: %s', msg)
+    log.error('{}: {}'.format(error, msg))
     raise exception_class(msg)
```

Comment:
This evaluates the expression every time, even if errors are suppressed!

Yes, I know in practice that this won't happen, but you should do all error messages the same way - where the expression is late evaluated.

https://docs.python.org/3.4/library/logging.html#logging.Logger.debug has the skinny here.

Sorta lame that it still uses old-style formatting strings, but the "correct" mechanism is one of these two:

    log.error('%s: %s', error, msg)

or

    log.error('%{error}: %{msg}', extra=locals())

(My preferred one because all the action happens in the format string...)